loader.js

```diff
@@ -60,14 +60,32 @@ function scanContent(options, content) {
 async function processLayout(options, frontMatter, content, resourcePath, scans) {
   const { mdxEnhancedPluginOptions: pluginOpts } = options
 
+  const extendedFm = await extendFrontMatter({
+    content,
+    frontMatter: {
+      ...frontMatter,
+      __resourcePath: resourcePath,
+      __scans: scans
+    },
+    phase: 'loader',
+    extendFm: pluginOpts.extendFrontMatter
+  })
+
+  const mergedFrontMatter = {
+    ...frontMatter,
+    ...extendedFm,
+      __resourcePath: resourcePath,
+      __scans: scans
+  }
```

Comment:
This consolidation is useful because it prevents potential bugs that could be introduced by forgetting to merge all the objects/properties. Or doing them in the incorrect order. I found a minor bug like that below.